Final_Submission1.js

```diff
@@ -0,0 +1,491 @@
+    console.log(
+      `The door slides open with in a slow rumble and finishes the process with a resounding BOOM!\n${wizardIcon}\nWizard: The fate of the kingdom rests in your hands now ${playerName}!\nYou step inside the cave only to be engulfed by darkness...\n`
+    );
+    answer = await ask(
+      `${playerName} cannot see at all! Check your inventory for tools you use.\n>_`
+    );
+  }
+  //Answer Sanitization
+  answer = answer.toLowerCase();
+  //Narrow Passage Failure Loop 1
+  while (!answer.includes("check inventory")) {
+    answer = await ask(
+      `${playerName} cannot see at all! Check your inventory for tools you use.\n>_`
+    );
+    answer = answer.toLowerCase();
+  }
+  //Answer Sanitization
+  answer = answer.toLowerCase();
+  //Narrow Passage Inventory Check
+  if (answer.includes("check inventory")) {
+    answer = await ask(`${player.inventory()}\n>_`);
+  }
+  //Answer Sanitization
+  answer = answer.toLowerCase();
+  //Narrow Passage Puzzle Failure Loop
+  while (!answer.includes("light torch")) {
+    answer = await ask(
+      `${playerName} cannot see at all! ${playerName} looks for their torch to light...\n>_`
+    );
+    answer = answer.toLowerCase();
+  }
+  //Answer Sanitization
+  answer = answer.toLowerCase();
+  //Narrow Passage Puzzle Success
+  if (answer.includes("light torch")) {
+    console.log(
+      `The room blazes with illumination as the torch catches fire!\n>_`
+    );
+    answer = await ask(currentRoom.displayDescription());
+  }
+  //Answer Sanitization
+  answer = answer.toLowerCase();
+  //Narrow Passage Puzzle Failure Loop
+  while (!answer.includes("crawl floor")) {
+    answer = await ask(
+      `The only way ${playerName} can progress is to crawl along the floor!\n>_`
+    );
+
+    answer = answer.toLowerCase();
+  }
+  //Answer Sanitization
+  answer = answer.toLowerCase();
+  //Narrow Passage Puzzle Success
+  //Treasure Room Transition
+  if (answer.includes("crawl floor")) {
+    currentRoom.treasureRoom();
+    console.log(
+      `${playerName} barely fits but manages to make it past a prolonged and suffocating squeeze!\nThe small opening drops ${playerName} down into the next room, too high to climb back out.\n>_`
+    );
+    answer = await ask(currentRoom.displayDescription());
+  }
+  //Answer Sanitization
+  answer = answer.toLowerCase();
+  //Player Adds Key to Inventory
+  while (!answer.includes("open door")) {
+    if (answer.includes("take key")) {
+      player.equip("key");
+      answer = await ask(`${player.inventory()}\n>_`);
+    } else {
+      answer = await ask(
+        `The only way ${playerName} can progress is to open the door!\n>_`
+      );
+    }
+    answer = answer.toLowerCase();
+  }
+  //Answer Sanitization
+  answer = answer.toLowerCase();
+
+  //Graveyard Room method without key
+  if (answer.includes("open door")) {
+    currentRoom.graveyardRoom();
+    answer = await ask(currentRoom.displayDescription());
+  }
```

Comment:
You could also add in the functionality at this room to be able to go back to the treasure room in order to try to find the key again